addons/event_crm/models/event_registration.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class EventRegistration(models.Model):
+    _inherit = 'event.registration'
+
+    lead_ids = fields.Many2many('crm.lead', string='Leads', readonly=True, help="Leads generated from the registration.", copy=False)
+    leads_count = fields.Integer('# Leads', compute='_compute_leads_count', help="Counter for the leads linked to this registration")
+
+    @api.depends('lead_ids')
+    def _compute_leads_count(self):
+        for record in self:
+            record.leads_count = len(record.lead_ids)
+
+    @api.model
+    def create(self, vals):
+        registration = super(EventRegistration, self).create(vals)
+        self.env['event.crm.rule'].search([])._create_leads(registration)
+        return registration
+
+    def write(self, vals):
+        registration = super(EventRegistration, self).write(vals)
+        for lead in self.lead_ids:
```

Comment:
Why this update ? If some pieces have to be updated, you should ensure they have been modified. And I am not sure it should be done at all, what is the purpose ?